src/components/inputs/date/date.component.tsx

```diff
@@ -29,8 +30,9 @@ const DateField: React.FC<FormFieldProps> = ({ question, onChange, handler, prev
     return false;
   }, [encounterContext.sessionMode, question.readonly, question.inlineRendering, layoutType, workspaceLayout]);
 
-  const onDateChange = ([date]) => {
-    const refinedDate = date instanceof Date ? new Date(date.setHours(0, 0, 0, 0)) : new Date(date);
+  const onDateChange = (date: CalendarDate) => {
+    const refinedDate = new Date(date.toString());
```

Comment:
```suggestion
    const refinedDate = new Date(date.toDate(getLocalTimeZone())));
```

(You may have to add the import)